src/test/java/BadVDiffTest.java

```diff
@@ -0,0 +1,43 @@
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.tinylog.Logger;
+import org.variantsync.diffdetective.diff.result.DiffParseException;
+import org.variantsync.diffdetective.util.StringUtils;
+import org.variantsync.diffdetective.variation.diff.DiffTree;
+import org.variantsync.diffdetective.variation.diff.bad.BadVDiff;
+import org.variantsync.diffdetective.variation.diff.graph.GraphView;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+public class BadVDiffTest {
+    private static final Path resDir = Constants.RESOURCE_DIR.resolve("bad");
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+            "1",
+            "const",
+            "diamond",
+            "deep_insertion"
+    })
+    public void toGood_after_fromGood_idempotency(String filename) throws IOException, DiffParseException {
+        final Path testfile = resDir.resolve(filename + ".diff");
+        Logger.debug("Testing " + testfile);
```

Comment:
These debug messages are meant to help the developer to debug any test failures. Knowing the parameters in JUnit won't help much because the object structure of variation diffs is hard to read. I used these log statements when developing the toGood and fromGood methods. I made them `debug` so that they won't appear in the default test output, only if we want to debug the tests. I would like to keep them for now because I have to work with bad diffs for the next two months and probably could need these logs.

Reply:
OK. I can agree to using `debug` for doing such logs.